pkg/compactor/compactor.go

```diff
@@ -41,8 +41,9 @@ const (
 )
 
 var (
-	errInvalidBlockRanges = "compactor block range periods should be divisible by the previous one, but %s is not divisible by %s"
-	RingOp                = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, nil)
+	errInvalidBlockRanges            = "compactor block range periods should be divisible by the previous one, but %s is not divisible by %s"
+	RingOp                           = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, nil)
+	WaitInstanceStateTimeoutDuration = time.Duration(600)
```

Comment:
This seems weird. `time.Duration(600)` is 600ns.
I recommend you make this `600*time.Second` (or perhaps `10*time.Minute`) and remove the multiplication on line 398.

Also why isn't it a constant?  (Maybe it should be configurable, but you didn't do that)

Reply:
Thanks for the review, I made changes to allow the timeout to be configurable.